src/eterna/puzzle/PuzzleManager.ts

```diff
@@ -149,6 +150,10 @@ export default class PuzzleManager {
             newpuz.nextPuzzleID = Number(json['next-puzzle']);
         }
 
+        if (json['next-puzzle-link']) {
+            newpuz.nextPuzzleLink = String(json['next-puzzle-link']);
```

Comment:
Elsewhere in this function we don't cast for values that are already strings
```suggestion
            newpuz.nextPuzzleLink = json['next-puzzle-link'];
```